lib/matplotlib/widgets.py

```diff
@@ -2062,6 +2075,56 @@ def set_handle_props(self, **handle_props):
             self.update()
         self._handle_props.update(handle_props)
 
+    def _validate_state(self, value):
+        supported_state = [
+            key for key, value in self._state_modifier_keys.items()
+            if key != 'clear' and value != 'not-applicable'
+            ]
+        if value not in supported_state:
+            keys = ', '.join(supported_state)
+            raise ValueError('Setting default state must be one of the '
```

Comment:
There's a helper function in `matplotlib.cbook` (or `matplotlib._api`?) called `check_in_list` which can be used here, which will automatically emit a nice error message.

Reply:
`_api`; that would become
```
check_in_list(supported_state, _state=value)
```
?